fix(PinInput): issues with deleting numeric values when type is number

J-Michalek opened the pull request:
Resolves: https://github.com/unovue/reka-ui/issues/1960

The issue was that an empty string is converted into 0 also deleting numeric values requires different handling than strings.

Commit:
fix(PinInput): issues with deleting numeric values when type is number

Commit:
fix(PinInput): allow 0 as valid value in numeric mode

zernonia reviewed (commented):
Can we have the test written for this case? 😁

We can also remove the computed `isNumericMode` in `PinInputInput`

Commit:
Merge branch 'v2' of https://github.com/J-Michalek/reka-ui into fix/pin-input-number

Commit:
test(pin-input): add more cases for type number

Commit:
refactor(pin-input-input): remove unused computed

J-Michalek commented:
Done, I have added more test cases https://github.com/unovue/reka-ui/pull/1965/commits/2e24cd78a4f7e1253db951bd15356c94936d055a and removed the unused computed.
